packages/typescript/core/src/core/api.ts

```diff
@@ -259,3 +259,167 @@ export function getLastActiveTraceId(): string | undefined {
   const traceManager = InMemoryTraceManager.getInstance();
   return traceManager.lastActiveTraceId;
 }
+
+/**
+ * Get the current active span in the global context.
+ *
+ * This only works when the span is created with fluent APIs like `@trace` or
+ * `withSpan`. If a span is created with the `startSpan` API, it won't be
+ * attached to the global context so this function will not return it.
+ *
+ * @returns The current active span if exists, otherwise undefined.
+ *
+ * @example
+ * ```typescript
+ * const tracedFunc = trace(() => {
+ *   const span = getCurrentActiveSpan();
+ *   span?.setAttribute("key", "value");
+ *   return 0;
+ * });
+ *
+ * tracedFunc();
+ * ```
+ */
+export function getCurrentActiveSpan(): LiveSpan | undefined {
+  const otelSpan = otelTrace.getActiveSpan();
+
+  // If no active span or it's a NonRecordingSpan, return undefined
+  if (!otelSpan || otelSpan.spanContext().traceId === '00000000000000000000000000000000') {
```

Comment:
Can we make `00000000000000000000000000000000` a const? It's easy to mistake the digits